## Zomato Reataurant Analysis: Data Understanding
**Objective:** Understanding the structure and quality of the dataset before analysis.
**Author:** Varish Ali

## Libraries

In [2]:
import pandas as pd
import numpy as np

## Data Load

In [7]:
df = pd.read_csv("C:/Users/hp/VR-PROJECTS/zomato-restaurant-analysis/data/zomato.csv",encoding="latin-1")

In [11]:
print("Rowa, Columns: ",df.shape)
print(df.head(5))

Rowa, Columns:  (9551, 21)
   Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Man

## Structure Check

In [13]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9551 entries, 0 to 9550
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   object 
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   object 
 4   Address               9551 non-null   object 
 5   Locality              9551 non-null   object 
 6   Locality Verbose      9551 non-null   object 
 7   Longitude             9551 non-null   float64
 8   Latitude              9551 non-null   float64
 9   Cuisines              9542 non-null   object 
 10  Average Cost for two  9551 non-null   int64  
 11  Currency              9551 non-null   object 
 12  Has Table booking     9551 non-null   object 
 13  Has Online delivery   9551 non-null   object 
 14  Is delivering now     9551 non-null   object 
 15  Switch to order menu 

,Restaurant ID,Country Code,Longitude,Latitude,Average Cost for two,Price range,Aggregate rating,Votes
count,9.551000e+03,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000,9551.000000
mean,9.051128e+06,18.365616,64.126574,25.854381,1199.210763,1.804837,2.666370,156.909748
std,8.791521e+06,56.750546,41.467058,11.007935,16121.183073,0.905609,1.516378,430.169145
min,5.300000e+01,1.000000,-157.948486,-41.330428,0.000000,1.000000,0.000000,0.000000
25%,3.019625e+05,1.000000,77.081343,28.478713,250.000000,1.000000,2.500000,5.000000
50%,6.004089e+06,1.000000,77.191964,28.570469,400.000000,2.000000,3.200000,31.000000
75%,1.835229e+07,1.000000,77.282006,28.642758,700.000000,2.000000,3.700000,131.000000
max,1.850065e+07,216.000000,174.832089,55.976980,800000.000000,4.000000,4.900000,10934.000000


## Missing Value Percentage Ke Sath

In [15]:
missing = df.isnull().sum()
missing

Restaurant ID           0
Restaurant Name         0
Country Code            0
City                    0
Address                 0
Locality                0
Locality Verbose        0
Longitude               0
Latitude                0
Cuisines                9
Average Cost for two    0
Currency                0
Has Table booking       0
Has Online delivery     0
Is delivering now       0
Switch to order menu    0
Price range             0
Aggregate rating        0
Rating color            0
Rating text             0
Votes                   0
dtype: int64

In [16]:
missing_pct = (missing / len(df) * 100).round(2)
missing_pct

Restaurant ID           0.00
Restaurant Name         0.00
Country Code            0.00
City                    0.00
Address                 0.00
Locality                0.00
Locality Verbose        0.00
Longitude               0.00
Latitude                0.00
Cuisines                0.09
Average Cost for two    0.00
Currency                0.00
Has Table booking       0.00
Has Online delivery     0.00
Is delivering now       0.00
Switch to order menu    0.00
Price range             0.00
Aggregate rating        0.00
Rating color            0.00
Rating text             0.00
Votes                   0.00
dtype: float64

In [19]:
pd.DataFrame({"missing":  missing, "percent": missing_pct}).query("missing > 0")

,missing,percent
Cuisines,9,0.09


## Duplicate And Unique Values

In [36]:
print("Duplicate rows:", df.duplicated().sum())

Duplicate rows: 0


In [37]:
print("Unique restaurants: ",df['Restaurant ID'].nunique())

Unique restaurants:  9551


In [38]:
print("Countries: ",df['Country Code'].nunique())

Countries:  15


In [41]:
print("Cities: ",df['City'].nunique())

Cities:  141


## Filter India

In [33]:
india = df[df['Country Code'] == 1].copy()
print("India restaurants: ",india.shape[0])
india['City'].value_counts().head(10)

India restaurants:  8652


City
New Delhi       5473
Gurgaon         1118
Noida           1080
Faridabad        251
Ghaziabad         25
Ahmedabad         21
Guwahati          21
Lucknow           21
Bhubaneshwar      21
Amritsar          21
Name: count, dtype: int64

## Important Columns

In [34]:
print(df.columns.tolist())

['Restaurant ID', 'Restaurant Name', 'Country Code', 'City', 'Address', 'Locality', 'Locality Verbose', 'Longitude', 'Latitude', 'Cuisines', 'Average Cost for two', 'Currency', 'Has Table booking', 'Has Online delivery', 'Is delivering now', 'Switch to order menu', 'Price range', 'Aggregate rating', 'Rating color', 'Rating text', 'Votes']


In [35]:
print(india['Cuisines'].value_counts().head(10))

Cuisines
North Indian                      936
North Indian, Chinese             511
Fast Food                         348
Chinese                           340
North Indian, Mughlai             334
Cafe                              279
Bakery                            216
North Indian, Mughlai, Chinese    197
Bakery, Desserts                  170
Street Food                       149
Name: count, dtype: int64


## Key Observations
- Dataset: 9,551 rows, 21 columns.
- Missing values: only `Cuisines` (9 rows, 0.09%).
- India has 8,652 restaurants; ~92% are in Delhi NCR.
- Other Indian cities have very few records (~21 each), so city-level comparison is unreliable.
- Plan: focus on NCR locality-level analysis.